test/evp_test.c

```diff
@@ -2254,8 +2259,12 @@ static int pkey_kdf_test_init(EVP_TEST *t, const char *name)
         return 0;
     kdata->ctx = EVP_PKEY_CTX_new_id(kdf_nid, NULL);
     if (kdata->ctx == NULL) {
-        OPENSSL_free(kdata);
-        return 0;
+        /*
+         * Don't fail immediately. This algorithm might not be available in this
```

Comment:
This line looks bad to me -  if the ctx is NULL, pkey_kdf_test_parse() would fall over..